agent/api/statechange.go

```diff
@@ -80,24 +90,69 @@ func (c *ContainerStateChange) String() string {
 	return res
 }
 
+// SetTaskTimestamps adds the timestamp information of task into the event
+// to be sent by SubmitTaskStateChange
+func (change *TaskStateChange) SetTaskTimestamps() {
+	if change.Task == nil {
+		return
+	}
+
+	// Send the task timestamp if set
+	if timestamp := change.Task.GetPullStartedAt(); !timestamp.IsZero() {
+		change.PullStartedAt = aws.Time(timestamp.UTC())
+	}
+	if timestamp := change.Task.GetPullStoppedAt(); !timestamp.IsZero() {
+		change.PullStoppedAt = aws.Time(timestamp.UTC())
+	}
+	if timestamp := change.Task.GetExecutionStoppedAt(); !timestamp.IsZero() {
+		change.ExecutionStoppedAt = aws.Time(timestamp.UTC())
+	}
+}
+
+// ShouldBeReport checks if the statechange should be reported to backend
+func (change *TaskStateChange) ShouldBeReport() bool {
```

Comment:
ShouldReport or ShouldBeReported

But I think we already have a way of determining whether state changes should be reported or not.  Can you explain what's different about this logic?

Reply:
The logic today only allow "RUNNING" and "STOPPED" status to be sent for task state change. While we also want to send the container change if the task is in "PENDING", eg container failed to create.